src/main/java/ladysnake/requiem/common/block/RunicObsidianBlockEntity.java

```diff
@@ -0,0 +1,307 @@
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.particle.ParticleTypes;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.tag.BlockTags;
+import net.minecraft.util.Tickable;
+import net.minecraft.util.math.*;
+import net.minecraft.world.BlockView;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+public class RunicObsidianBlockEntity extends BlockEntity implements Tickable {
+    public static final Direction[] OBELISK_SIDES = {Direction.SOUTH, Direction.EAST, Direction.NORTH, Direction.WEST};
+    public static final int POWER_ATTEMPTS = 1;
+    private final Object2IntMap<StatusEffect> levels = new Object2IntOpenHashMap<>();
+    private int obeliskWidth = 0;
+    private int obeliskHeight = 0;
+
+    public RunicObsidianBlockEntity() {
+        super(RequiemBlockEntities.RUNIC_OBSIDIAN);
+    }
+
+    @Override
+    public void tick() {
+        assert this.world != null;
+        if (this.world.isClient) return;
+
+        if (this.world.getTime() % 80L == 0L) {
+            this.refresh();
+
+            Vec3d obeliskCenter = new Vec3d(
+                MathHelper.lerp(0.5, this.pos.getX(), this.pos.getX() + obeliskWidth - 1),
+                this.pos.getY() - 2,
+                MathHelper.lerp(0.5, this.pos.getZ(), this.pos.getZ() + obeliskWidth - 1)
+            );
+
+            if (!this.levels.isEmpty() && this.findPowerSource((ServerWorld) this.world, obeliskCenter, this.obeliskWidth * 5)) {
+                this.applyPlayerEffects();
+            }
+        }
+    }
+
+    private void applyPlayerEffects() {
+        Preconditions.checkState(this.world != null);
+        Preconditions.checkState(this.obeliskWidth > 0);
+        Preconditions.checkState(this.obeliskHeight > 0);
+
+        double range = this.obeliskWidth * 10 + 10;
+        Box box = (new Box(this.pos, this.pos.add(obeliskWidth-1, obeliskHeight-1, obeliskWidth-1))).expand(range);
+
+        int effectDuration = (9 + this.obeliskWidth * 2) * 20;
+        List<PlayerEntity> players = this.world.getNonSpectatingEntities(PlayerEntity.class, box);
+
+        for (PlayerEntity player : players) {
+            if (RemnantComponent.get(player).getRemnantType().isDemon()) {
+                for (Object2IntMap.Entry<StatusEffect> effect : this.levels.object2IntEntrySet()) {
+                    player.addStatusEffect(new StatusEffectInstance(effect.getKey(), effectDuration, effect.getIntValue() - 1, true, true));
+                }
+            }
+        }
+    }
+
+    private boolean findPowerSource(ServerWorld world, Vec3d center, double range) {
+        BlockPos.Mutable checked = new BlockPos.Mutable();
+
+        for (int attempt = 0; attempt < RunicObsidianBlockEntity.POWER_ATTEMPTS; attempt++) {
+            // https://stackoverflow.com/questions/5837572/generate-a-random-point-within-a-circle-uniformly/50746409#50746409
```

Comment:
Should this be within a circle? Or a square, since that's (very mildly) faster, and square

Reply:
Circle looks prettier 🙃

Reply:
fair enough!